core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -843,9 +847,9 @@ private[spark] class ExecutorAllocationManager(
     def removeStageFromResourceProfileIfUnused(stageAttempt: StageAttempt): Unit = {
       if (!stageAttemptToNumRunningTask.contains(stageAttempt) &&
           !stageAttemptToNumTasks.contains(stageAttempt) &&
-          !stageAttemptToNumSpeculativeTasks.contains(stageAttempt) &&
-          !stageAttemptToTaskIndices.contains(stageAttempt) &&
-          !stageAttemptToSpeculativeTaskIndices.contains(stageAttempt)
+          !stageAttemptToSpeculativeTaskIndices.contains(stageAttempt) &&
```

Comment:
hmm... why does `!stageAttemptToSpeculativeTaskIndices.contains(stageAttempt)` move from after `!stageAttemptToTaskIndices.contains(stageAttempt)` to before  `!stageAttemptToTaskIndices.contains(stageAttempt)`?